x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java

```diff
@@ -5,141 +5,107 @@
-import org.elasticsearch.action.search.SearchRequest;
-import org.elasticsearch.common.bytes.BytesArray;
-import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.common.xcontent.XContentHelper;
-import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.query.QueryBuilders;
-import org.elasticsearch.search.builder.SearchSourceBuilder;
-import org.elasticsearch.xpack.core.ml.action.DeleteTrainedModelAction;
-import org.elasticsearch.xpack.core.ml.action.PutTrainedModelAction;
+import org.elasticsearch.test.SecuritySettingsSourceField;
+import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.xpack.core.ml.inference.MlInferenceNamedXContentProvider;
-import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
+import org.elasticsearch.xpack.core.ml.integration.MlRestTestStateCleaner;
 import org.junit.After;
 import org.junit.Before;
 
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
+import static org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken.basicAuthHeaderValue;
 import static org.hamcrest.CoreMatchers.containsString;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.is;
 
-public class InferenceIngestIT extends MlNativeAutodetectIntegTestCase {
+public class InferenceIngestIT extends ESRestTestCase {
+
+    private static final String BASIC_AUTH_VALUE_SUPER_USER =
+        basicAuthHeaderValue("x_pack_rest_user", SecuritySettingsSourceField.TEST_PASSWORD_SECURE_STRING);
 
     @Before
     public void createBothModels() throws Exception {
-        client().execute(PutTrainedModelAction.INSTANCE, new PutTrainedModelAction.Request(buildClassificationModel())).actionGet();
-        client().execute(PutTrainedModelAction.INSTANCE, new PutTrainedModelAction.Request(buildRegressionModel())).actionGet();
+        Request request = new Request("PUT", "_ml/inference/test_classification");
+        request.setJsonEntity(CLASSIFICATION_CONFIG);
+        client().performRequest(request);
+
+        request = new Request("PUT", "_ml/inference/test_regression");
+        request.setJsonEntity(REGRESSION_CONFIG);
+        client().performRequest(request);
+    }
+
+    public void clearMlState() throws Exception {
```

Comment:
I think there should be a wider rethink of the cleanup methods.

`MlNativeAutodetectIntegTestCase` extends `MlNativeIntegTestCase`, and `MlNativeIntegTestCase` already has an `@After` method that calls the abstract method `cleanUpResources()` then waits for pending tasks.

Since this `clearMlState()` method also waits for pending tasks that means we're waiting twice, which will slow down the cleanup.  If possible I would prefer that we only wait for pending tasks once, at the very end of all cleanup operations.

@benwtrent please can you try to integrate your cleanup into the `MlNativeIntegTestCase` framework, by overriding `cleanUpResources()`.  And if that doesn't work for some reason and multiple cleanup phases are required please add some comments explaining what the full sequence is and why a simpler sequence doesn't work.  (For example, I can imagine there might be something annoying like the ML annotations index being created part way through the cleanup, but if you found that problem and worked around it it would be good to document that this can happen so that the next maintainer can take it into account.)

Reply:
@droberts195 I don't understand 

> Since this clearMlState() method also waits for pending tasks that means we're waiting twice, which will slow down the cleanup. 

This test class is now inheriting from `ESRestTestCase` which does not wait for tasks to finish. 

This clean up code is the same we use for other `ESRestTestCase` classes. 

`MlNativeIntegTestCase` already does something similar (see: https://github.com/elastic/elasticsearch/blob/d2aee62468ae35d5fe95ffc1161a0770148af2e0/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeIntegTestCase.java#L128) 

As for the issues I originally had that prompted this change to `ESRestTestCase`, the issue has nothing to do with any code paths that the ML team has written. It even occurred with the `set` processor. So, adjusting our cleanup process would do nothing.

@przemekwitek 

> Do you plan to use this method in other places?

I could make it private

Reply:
Sorry, I must have been colour blind when reading the diff.